### ISSUE TITLE
Missing messages in GUI

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#298]
**Reported by KSMarksPsych on 23 Oct 37507395 02:40 UTC**
First reported [here](http://boinc.berkeley.edu/dev/forum_thread.php?id=1936&nowrap=true#11335), and a second report in the same thread, [here](http://boinc.berkeley.edu/dev/forum_thread.php?id=1936&nowrap=true#11512).

There seem to be large gaps with missing messages in the message window/tab of the GUI but all messages appear in stdoutdae.txt.

Reported as happening on both 5.10.7 and 5.10.8.

**Comment by romw** [comment]
**Commented by WHRoeder on 20 Apr 37507747 17:46 UTC**
Replying to [KSMarksPsych](ticket:306):
[the 5.10.8 logs.
[[BR]([BR]]Here's)]GUI lines 54-57:
 ```
7/3/2007 10:35:51 PM|SETI@home|Restarting task 26fe00aa.9890.19122.704808.3.122_0 using setiathome_enhanced version 515
7/3/2007 10:35:51 PM||[enforce_schedule: end
7/4/2007 6:52:09 AM||[cpu_sched_debug](cpu_sched_debug]) finished preempt loop, nrunning 1
7/4/2007 6:52:09 AM|Einstein@Home|[h1_0429.05_S5R2__82_S5R2c_0 sched state 1 next 1 task state 0
 ```
[[BR](cpu_sched_debug])]stdoutdae.txt lines 3659-3663
 ```
2007-07-03 22:35:51 [Restarting task 26fe00aa.9890.19122.704808.3.122_0 using setiathome_enhanced version 515
2007-07-03 22:35:51 [---](SETI@home]) [enforce_schedule: end
2007-07-03 22:36:51 [---](cpu_sched_debug]) [enforce_schedule(): start
2007-07-03 22:36:51 [SETI@home](cpu_sched_debug]) [want to run: 26fe00aa.9890.19122.704808.3.122_0
 ```
[[BR](cpu_sched_debug])]stdoutdae.txt lines 8307-8314
 ```
2007-07-04 06:51:09 [[cpu_sched_debug](SETI@home]) 26fe00aa.9890.19122.704808.3.127_1 sched state 2 next 2 task state 1
2007-07-04 06:51:09 [[cpu_sched_debug](---]) enforce_schedule: end
2007-07-04 06:52:09 [[cpu_sched_debug](---]) enforce_schedule(): start
2007-07-04 06:52:09 [[cpu_sched_debug](SETI@home]) want to run: 26fe00aa.9890.19122.704808.3.127_1
2007-07-04 06:52:09 [[cpu_sched_debug](SETI@home]) processing 26fe00aa.9890.19122.704808.3.127_1
2007-07-04 06:52:09 [[cpu_sched_debug](SETI@home]) 26fe00aa.9890.19122.704808.3.127_1 is already running
2007-07-04 06:52:09 [[cpu_sched_debug](---]) finished preempt loop, nrunning 1
2007-07-04 06:52:09 [[cpu_sched_debug](Einstein@Home]) h1_0429.05_S5R2__82_S5R2c_0 sched state 1 next 1 task state 0
 ```

**Comment by romw** [comment]
**Commented by WHRoeder on 29 Mar 37542879 23:33 UTC**
> Reported as happening on both 5.10.7 and 5.10.8.
5.10.13 also

**Comment by romw** [comment]
**Commented by Didactylos on 16 Jul 37879690 21:46 UTC**
Please attach large logs as files. Deleted comment is reproduced here:
---
08/22/07 19:54:33 changed by Barbud

I was the original complainer about this message some time ago. I only noticed it because I am doing alpha testing sometimes and I would leave the computer for a few days. When I come back to look back on the messages via the GUI I noticed large chunks of time were missing. I would then have to go directly to the stdoutdae file to see if there any errors.

This pc used to be a high end gaming pc now it just runs BOINC.[E6700 processor, waiting for the wife to say ok to buy the quad.[[BR]([BR]]
Intel)]
4Gig Ram[Home[[BR]([BR]]
WINXP)]
BOINC 5.10.13

I have copied the entire contents of the GUI messages tab. Sorry it is so long and I am not sure how to make the scroll bar for the code block smaller. As you can see by scanning thru the dates there are large blocks missing. Now if I go into the stdoutdae file everything is there except for some of the first entries in the GUI messages are not int the stdoutdae. Below are the first entries in the stdoutdae.txt but if you look in the next block these are not even listed in the GUI messages tab.
First lines from stdoutdae.txt
```
2007-08-17 15:57:16 [Lattice Project](The) [Finished upload of file 224482590.9780102250632569.23_2_3
2007-08-17 15:57:16 [The Lattice Project](file_xfer]) [file_xfer] Throughput 4074 bytes/sec
```

BOINC 5.10.13 GUI messages

<snip massive log>

This is repeatable. I can send files to anybody that wants to look closer. Please advise. Again sorry that this message is so long.

Thanks for the assistance.

**Comment by romw** [comment]
**Commented by Didactylos on 15 Feb 38482710 12:53 UTC**
Explanation by Rom (from boinc_alpha):

''After 20 minutes or so Windows shuts down the timers the manager uses to maintain state, so it stops getting the latest message updates.  The CC only retains the last 1000 messages in memory, so when a bunch of log flags are turned on that buffer gets cycled through really quickly.  When the manager wakes back up it requests all the latest messages and gets back the last 1000, and anything before that is forgotten.''

**Comment by romw** [comment]
**Commented by romw on 17 Mar 38482910 05:46 UTC**
This will be resolved with the changes related to the RPC Thread in the 6.4 release.

**Comment by romw** [comment]
**Commented by romw on 7 Apr 38482912 13:20 UTC**
I should also point out that this bug has been around since 06/2006.

**Comment by romw** [comment]
**Commented by romw on 21 Nov 38498957 12:53 UTC**
Updated from last message received about issue.

----- Rom

> -----Original Message-----
> From: David Anderson [
> Sent: Tuesday, June 24, 2008 1:19 PM
> To: Rom Walton
> Cc: Jorden van der Elst; Charlie Fenton
> Subject: Re: [boinc_alpha](mailto:davea@ssl.berkeley.edu]) BOINC 6.2.8 released for Windows
>
>The bottom line is:
>
>- we need to poll for messages no matter what tab is open
>- we need to use something other than WM_TIMER.
>
>Let's defer this until we put in the GUI RPC thread
>(and let's get started with that ASAP)
>
>-- David

**Comment by romw** [comment]
**Commented by romw on 10 Jul 38766791 01:46 UTC**
This should now be fixed in the 6.3.x builds.

**Comment by romw** [comment]
**Commented by Ageless on 29 Aug 38778021 14:40 UTC**
Reopening. It's not fixed in 6.3.14
```
09-Oct-08 13:44:11|SETI@home|[result 19au08aa.30605.10706.11.8.83_1 checkpointed
09-Oct-08 13:45:12|SETI@home|[checkpoint_debug](checkpoint_debug]) result 19au08aa.30605.10706.11.8.83_1 checkpointed
09-Oct-08 13:46:05|Einstein@Home|[result h1_1103.35_S5R4__1013_S5R4a_1 checkpointed
09-Oct-08 18:26:31||[file_xfer_debug](checkpoint_debug]) URL: http://www.primegrid.com/download/llrTPS_workunit.php?id=18483756
09-Oct-08 18:26:33||[FILE_XFER_SET::poll(): http op done; retval 0
09-Oct-08 18:26:33||[file_xfer_debug](file_xfer_debug]) file transfer status 0
09-Oct-08 18:26:33|PrimeGrid|Finished download of llrTPS_18483756
```

and 

```
10-Oct-08 01:00:28|Einstein@Home|[result h1_1103.35_S5R4__1013_S5R4a_1 checkpointed
10-Oct-08 01:01:45||Running CPU benchmarks

10-Oct-08 01:02:18||Resuming computation
10-Oct-08 05:58:28|SETI@home|[checkpoint_debug](checkpoint_debug]) result 19au08ab.6780.3753.5.8.196_1 checkpointed
10-Oct-08 05:59:33|SETI@home|[result 19au08ab.6780.3753.5.8.196_1 checkpointed
10-Oct-08 06:00:39|SETI@home|[checkpoint_debug](checkpoint_debug]) result 19au08ab.6780.3753.5.8.196_1 checkpointed
10-Oct-08 06:00:39|PrimeGrid|[checkpoint_debug] result llrTPS_18492261_0 checkpointed
```

**Comment by romw** [comment]
**Commented by Pepo on 23 May 38839038 09:20 UTC**
I've been today connected with Windows 6.3.17 Manager to remote Windows 6.3.19 client. The remote machine went offline for some time (suspended for nearly a hour). Half day later I've noticed that the Messages page is not being updated at all, approximately since the off-line period. I've tried to switch into Simple and back to Advanced view - the list was finished with the same old messages. All other tabs were live and up-to-date.

Selecting the host again caused all messages to be reread, the tab was being updated afterwards.

**Comment by romw** [comment]
**Commented by charlief on 25 Aug 38947022 04:53 UTC**
This has been fixed as of version 6.3.15.

**Comment by romw** [comment]
**Commented by Nicolas on 13 Mar 38947710 21:20 UTC**
Why do you bump the milestone if it's already fixed?

**Comment by romw** [comment]
**Commented by charlief on 30 Jul 38948281 00:26 UTC**
Replying to [Nicolas](comment:16):
> Why do you bump the milestone if it's already fixed?
Because, as I'm sure you know, 6.4 was not built on 6.3.x (SVN trunk), it was built on 6.2.x so it is _not_ fixed in 6.4.  6.3 has been renumbered as 6.5.x, so the fix will be released as 6.6.x.
